pkg/acsclient/client.go

```diff
@@ -0,0 +1,124 @@
+package acsclient
+
+import (
+	"fmt"
+	"net/http"
+)
+
+var (
+	apiVersion = "v1"
+)
+
+type Query map[string]string
+type Client interface {
+	GetClusters() ([]Cluster, error)
+	GetNamespaces(query Query) ([]Namespace, error)
+	GetImages(query Query) ([]Image, error)
+	GetExportDeployments(query Query) ([]ExportDeployment, error)
+	GetExportImages(query Query) ([]ExportImage, error)
+}
+
+type acsClient struct {
+	Endpoint string
+	Token    string
+	timeout  int
+}
+
+func NewACSClient(endpoint, token string, timeout int) Client {
+	return &acsClient{
+		Endpoint: endpoint,
+		Token:    token,
+		timeout:  timeout,
+	}
+}
+
+func (client *acsClient) GetClusters() ([]Cluster, error) {
+	clustersURL := fmt.Sprintf("%s/%s/clusters", client.Endpoint, apiVersion)
+	var clustersResponse ClustersResponse
+	err := makeRequest(requestParams{
+		token:   client.Token,
+		url:     clustersURL,
+		method:  http.MethodGet,
+		timeout: client.timeout,
+	}, &clustersResponse)
+	if err != nil {
+		return nil, err
+	}
+	return clustersResponse.Clusters, nil
+}
+
+func (client *acsClient) GetNamespaces(query Query) ([]Namespace, error) {
+	namespacesURL := fmt.Sprintf("%s/%s/namespaces", client.Endpoint, apiVersion)
+	if query != nil {
+		namespacesURL += fmt.Sprintf("?query.query=%s", getQueryString(query))
+	}
```

Comment:
should query string constructed using https://pkg.go.dev/net/url#Values?

Reply:
can we use standard net/url to handle all url build instead of `Sprintf`, like

```golang
        namespacesURL, err := url.JoinPath(client.Endpoint, apiVersion, "namespaces)
	if query != nil {
		queryParams := namespacesURL.Query()
		queryParams.Set("query.query", getQueryString(query))
		namespacesURL.RawQuery = queryParams.Encode()
	}
```